pkg/gcrane/gc.go

```diff
@@ -17,62 +17,182 @@ package gcrane
 import (
 	"fmt"
 	"log"
+	"net/http"
+	"time"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/spf13/cobra"
 )
 
 func init() { Root.AddCommand(NewCmdGc()) }
 
 func NewCmdGc() *cobra.Command {
+	before := ""
+	after := ""
+
 	recursive := false
+	remove := false
+	untag := false
+
 	cmd := &cobra.Command{
 		Use:   "gc",
-		Short: "List images that are not tagged",
+		Short: "Garbage collect images",
 		Args:  cobra.ExactArgs(1),
 		Run: func(_ *cobra.Command, args []string) {
-			gc(args[0], recursive)
+			opts, err := parseOpts(before, after, recursive, remove, untag)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			if err := GarbageCollect(args[0], opts); err != nil {
+				log.Fatalln(err)
+			}
 		},
 	}
 
+	// Filters
+	cmd.Flags().StringVarP(&before, "before", "b", "", "Match images uploaded before this time (RFC3339)")
+	cmd.Flags().StringVarP(&after, "after", "a", "", "Match images uploaded after this time (RFC3339)")
+
+	// Behaviors
 	cmd.Flags().BoolVarP(&recursive, "recursive", "r", false, "Whether to recurse through repos")
+	cmd.Flags().BoolVarP(&remove, "delete", "D", false, "Delete images instead of just printing them")
+	cmd.Flags().BoolVarP(&untag, "untag", "U", false, "Also untag matching images")
 
 	return cmd
 }
 
-func gc(root string, recursive bool) {
+type GCOptions struct {
+	Before *time.Time
+	After  *time.Time
+
+	Recursive bool
+	Delete    bool
+	Untag     bool
+}
+
+// GarbageCollect iterates over a repository, printing and optionally deleting
+// images that match the criteria in opts.
+func GarbageCollect(root string, opts GCOptions) error {
```

Comment:
How about `pkg/v1/google.GarbageCollect`? This effectively just calls methods available in other packages and in `google` but provides a higher-level all-in-one method.

Reply:
That's kind of why I like it in `pkg/gcrane`, since that's exactly what gcrane does. `GarbageCollect` is a lot higher level thing than the primitives in `pkg/v1/google` and `pkg/v1/remote`, so I don't _really_ think it belongs there.